README.md

```diff
@@ -39,3 +39,69 @@ Most probably, you will have to adjust these four files:
+``` 
+mvn javadoc:aggregate
+```
+inside the Kitodo.Production main folder. This call produces the documentation inside the default directory beneath *${project.basedir}/target/site*. 
+
+To transport the generated file to the correct place (*${project.basedir}/docs*) issue
+```
+mvn --non-recursive antrun:run
+```
+
+Afterwards the contents of the "docs" directory may be committed.
+
+To make sure that Javadocs are up to date at the time of the committing the call of
+```
+mvn clean
+```
+also deletes the contents of _JavaDocs/JavaDocsGenerationDir_. This makes sure that the directory does not contain parts of files 
+that came into existence during earlier calls of the generation. To get a fresh Javadoc you may issue subsequently:
+
+```
+mvn clean
+mvn javadoc:aggregate 
+mvn --non-recursive antrun:run
+```
+Be careful on committing. Look twice what you commit. It may be the case that you checked out the JavaDocs and deleted them
+during the build process locally. It is possible you don't want to commit them (as they are locally deleted).
+
+#### Javadoc is graceful
+The configuration of the Javadoc-plugin is set to be graceful. The config tells Javadoc not to stop if there is an error and not
+to fail if there was one. This gives the developer the chance to generate "incorrect" Javadocs which also includes
+Javadocs that are incomplete (for example because of missing docs for a parameter of a function). 
+
+Changing these parameters may result in Javadocs of a higher quality as lots of documentation is requested but also may
+result in no documentation at all because the compilation just stops because of bad or insufficient Javadoc comments.
+
+```xml
+<configuration>
+	<!-- setting to true halts the generation at the first error -->
+	<failOnError>false</failOnError>
+	<!-- commenting this line enables the doclint checks on the sources - this 
+	     leads to fail with badly documented sources -->
+	<additionalparam>-Xdoclint:none</additionalparam>
+</configuration>
+```
+
+### Viewing documentation
+Generally there is the idea to generate a viewable form of the documentation automatically by ReadTheDocs. There is a project set 
+up (https://readthedocs.org/projects/kitodo-production/) that has a connection to the Kitodo.Production development project.
+Under normal conditions any check in to the docs-path of that project should trigger a rebuild of the documentation.
+
+What if not?  
+
+#### Checking the documentation locally
+You may run MkDocs (the tool that is used by ReadTheDocs under the hood) locally as well. To do that 
+
+* install MkDocs on your machine by following the instruction mentioned [here](http://www.mkdocs.org/#installation)
+* change to the Kitodo.Production main directory (the one where the "docs" directory can be found)
+* run  "mkdocs serve" at the command line 
+* point your browser to http://127.0.0.1:8000/
+
+For more information check [this page](check this page http://www.mkdocs.org/#getting-started).
```

Comment:
Please check the generated output for this link (https://github.com/Ishiido/kitodo-production/blob/add_developer_docu/README.md) - I don't think that you want this.